src/psyclone/domain/lfric/lfric_stencils.py

```diff
@@ -0,0 +1,649 @@
+        :type arg: :py:class:`psyclone.dynamo0p3.DynKernelArgument`
+        :param str context: a context for this stencil (e.g. "size" or \
+                            "direction").
+
+        :returns: unique string identifying the stencil for this argument.
+        :rtype: str
+
+        :raises GenerationError: if an explicit stencil extent is found in \
+                                 the meta-data for the kernel argument.
+        '''
+        unique = context
+        unique += arg.function_space.mangled_name
+        unique += arg.descriptor.stencil['type']
+        if arg.descriptor.stencil['extent']:
+            raise GenerationError(
+                "Found a stencil with an extent specified in the metadata. "
+                "This is not coded for.")
+        unique += arg.stencil.extent_arg.text.lower()
+        if arg.descriptor.stencil['type'] == 'xory1d':
+            unique += arg.stencil.direction_arg.text.lower()
+        return unique
+
+    def map_name(self, arg):
+        '''
+        Creates and registers a name for the stencil map associated with the
+        supplied kernel argument.
+
+        :param arg: kernel argument with which the stencil is associated.
+        :type arg: :py:class:`psyclone.dynamo0p3.DynKernelArgument`
+
+        :returns: a valid unique map name for a stencil in the PSy layer.
+        :rtype: str
+        '''
+        root_name = arg.name + "_stencil_map"
+        unique = LFRicStencils.stencil_unique_str(arg, "map")
+        return self._symbol_table.find_or_create_tag(unique, root_name).name
+
+    @staticmethod
+    def dofmap_symbol(symtab, arg):
+        '''
+        Creates and registers a symbol for the stencil dofmap associated with
+        the supplied kernel argument.
+
+        :param symtab: symbol table that will contain (or already contains) \
+            the symbol with this name.
+        :type symtab: :py:class:`psyclone.psyir.symbols.SymbolTable`
+        :param arg: kernel argument with which the stencil is associated.
+        :type arg: :py:class:`psyclone.dynamo0p3.DynKernelArgument`
+
+        :returns: a dofmap symbol for a stencil in the PSy layer.
+        :rtype: :py:class:`psyclone.psyir.symbols.Symbol`
+
+        '''
+        root_name = arg.name + "_stencil_dofmap"
+        unique = LFRicStencils.stencil_unique_str(arg, "dofmap")
+        if arg.descriptor.stencil['type'] == "cross2d":
+            num_dimensions = 4
+        else:
+            num_dimensions = 3
+        return symtab.find_or_create_array(root_name, num_dimensions,
+                                           ScalarType.Intrinsic.INTEGER,
+                                           tag=unique)
+
+    @staticmethod
+    def dofmap_size_symbol(symtab, arg):
+        '''
+        Create a valid symbol for the size (in cells) of a stencil
+        dofmap in the PSy layer.
+
+        :param symtab: symbol table that will contain (or already contains) \
```

Comment:
Please remove the `\` character and align "the symbol" in the line below with "symbol table".